pkg/util/scrubber/scrubber.go

```diff
@@ -159,3 +168,38 @@ func (c *Scrubber) scrub(data []byte, replacers []Replacer) []byte {
 	}
 	return data
 }
+
+// gitleaks applies the gitleaks library to the given data.
+func (c *Scrubber) gitleaks(data []byte) []byte {
+	// get gitleaks' findings within this set of bytes
+	findings := c.detector.DetectBytes(data)
+	if len(findings) == 0 {
+		return data
+	}
+
+	// Split the data by line.  Note that this does not copy the original data,
```

Comment:
Is the assumption that the original data can be scrubbed in-place new or do the other scrubbers do the same? If it's the former, we may need to consider if there may be knock-on effects of this change.